index.js

```diff
@@ -47,10 +47,9 @@ function renderElements(timestamp) {
 
 /**
  * Please see README.md for usage information.
- *
  * @param {object} opts Custom element options
  *
- * @param {any}       opts.component  Svelte component instance to incorporate into your custom element.
+ * @param {import('svelte').SvelteComponent<{}>}   opts.component  Svelte component instance to incorporate into your custom element.
```

Comment:
Looking at Svelte's definition of `SvelteComponent`, there's more here than just `Props` (the first generic parameter here, `{}`). There are also `Events` and `Slots`, too.

Would it be best to leave the generic here blank (i.e. simply `{import('svelte').SvelteComponent}`) and then allow those to default to `any`? Presumably that'll work fine for you as well after rebuilding. LMK!

![image](https://github.com/patricknelson/svelte-retag/assets/4269377/05b83bb5-7ade-4b9e-9771-8b4082f7239a)

Upstream `SvelteComponent` type:

![image](https://github.com/patricknelson/svelte-retag/assets/4269377/eaec08f8-0ced-4c11-b9c6-0182eba8a324)

Reply:
c.f. https://svelte.dev/docs/typescript#types-componenttype

Reply:
Resolving convo since I think the issue is that we are just taking in a constructor here (i.e. `Newable`). See https://github.com/patricknelson/svelte-retag/pull/33#issuecomment-1843789833